timemachine/src/time_travel.rs

```diff
@@ -0,0 +1,742 @@
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use std::time::Duration;
+use std::time::SystemTime;
+
+use anyhow::Context;
+use anyhow::Result;
+use denokv_proto::backup::BackupKvMutationKind;
+use futures::FutureExt;
+use futures::StreamExt;
+use futures::TryStreamExt;
+use rusqlite::OptionalExtension;
+
+use crate::backup::DatabaseBackupSource;
+use crate::backup::MutationRangeEntry;
+use crate::backup::MutationRangeKey;
+use crate::backup::SnapshotRangeKey;
+use crate::backup_source_s3::DatabaseBackupSourceS3Config;
+use crate::key_metadata::KeyMetadata;
+
+const MIGRATIONS: [&str; 1] = ["
+create table migration_state(
+  k integer not null primary key,
+  version integer not null
+);
+create table data_version (
+  k integer primary key,
+  version integer not null,
+  seq integer not null default 0
+);
+insert into data_version (k, version) values (0, 0);
+create table kv_snapshot (
+  k blob primary key,
+  v blob not null,
+  v_encoding integer not null,
+  version integer not null,
+  seq integer not null default 0,
+  expiration_ms integer not null default -1
+) without rowid;
+create view kv as
+  select k, v, v_encoding, version, seq, expiration_ms from kv_snapshot;
+insert into migration_state (k, version) values (0, 3);
```

Comment:
yes

Reply:
If it doesn't, I'd remove this.

Reply:
I have changed `migration_state` to a view to prevent accidental migration. We may still want to check the migration state in the future (for example if we have a breaking change to `kv` table), so I'd keep this...